deepchecks/vision/checks/performance/single_dataset_scalar_performance.py

```diff
@@ -0,0 +1,146 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing a check for computing a scalar performance metric for a single dataset."""
+import numbers
+import typing as t
+import warnings
+
+import torch
+from ignite.metrics import Accuracy, Metric
+
+from deepchecks.core import CheckResult, ConditionResult, DatasetKind
+from deepchecks.core.condition import ConditionCategory
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.vision import Batch, Context, SingleDatasetCheck
+from deepchecks.vision.metrics_utils.object_detection_precision_recall import \
+    ObjectDetectionAveragePrecision
+from deepchecks.vision.vision_data import TaskType
+
+__all__ = ['SingleDatasetScalarPerformance']
+
+
+class SingleDatasetScalarPerformance(SingleDatasetCheck):
+    """Calculate a performance metric as a scalar for a given model and a given dataset.
+
+    Parameters
+    ----------
+        metric: Metric, default: None
+        An ignite.Metric object whose score should be used. If None is given, use the default metric.
+        reduce: torch function, default: None
+        The function to reduce the scores tensor into a single scalar. For metrics that return a scalar use None
+        (default).
+        metric_name: str, default: None
+        A name for the metric to show in the check results.
+        reduce_name: str, default: None
+        A name for the reduce function to show in the check results.
```

Comment:
No need to indent (it will look bad on the docs when indenting). Take a look at other checks.